packages/nodes-base/utils/sendAndWait/utils.ts

```diff
@@ -357,6 +357,12 @@ export async function sendAndWaitWebhook(this: IWebhookFunctions) {
 	const res = this.getResponseObject();
 	const req = this.getRequestObject();
 
+	const isTokenValid = this.validateExecutionWaitingToken();
```

Comment:
Would it be possible to do this in core, maybe in `waiting-webhooks`, for any waiting execution that is resumed?

Reply:
as for now doing this will break other waiting execution, because updates in nodes will be needed, this PR deals only with HiTL operations

Reply:
Ideally we should do this in core and provide necessary methods for nodes. I think it’s ok to do it for hitl nodes first

Reply:
I was thinking if we could always add the signature to `$execution.resumeUrl` and verify the signature of all (not specific nodes) incoming requests on `waiting-webhook`.

Then node code wouldn't have to be changed and `$execution.resumeUrl` would keep working for existing workflows.

Reply:
we could, but it would increase the scope
we’d have to ensure that the signature is always included in all possible waiting executions

changes to the nodes would still be required -  for example like in this PR for including url in service's message